LabExT/Movement/Calibration.py

```diff
@@ -45,6 +74,62 @@ def __str__(self) -> str:
         return self.name.replace('_', ' ').capitalize()
 
 
+class AxesRotation:
+    """
+    Assigns a stage axis (X,Y,Z) to the positive chip axes (X,Y,Z) with associated direction.
+    If the assignment is well defined, a rotation matrix is calculated, which rotates the given chip coordinate perpendicular to the stage coordinate.
+    The rotation matrix (3x3) is therefore a signed permutation matrix of the coordinate axes of the chip.
+
+    Each row of the matrix represents a chip axis. Each column of the matrix represents a stage axis.
+    """
+
+    def __init__(self) -> None:
+        self._n = len(Axis)
+        self._matrix = np.identity(len(Axis))  # 3x3 identity matrix
+
+    def update(self, chip_axis: Axis, stage_axis: Axis, direction: Direction):
+        """
+        Updates the axes rotation matrix.
+        Replaces the column vector of given chip with signed (direction) i-th unit vector (i is stage)
+        """
+        if not (isinstance(chip_axis, Axis) and isinstance(stage_axis, Axis)):
+            raise ValueError("Unknown axes given for calibration.")
+
+        if not isinstance(direction, Direction):
+            raise ValueError("Unknown direction given for calibration.")
+
+        # Replacing column of chip with signed (direction) i-th unit vector (i
+        # is stage)
+        self._matrix[:, chip_axis.value] = np.eye(
+            1, 3, stage_axis.value) * direction.value
+
+    @property
+    def is_valid(self):
+        """
+        Checks if given matrix is a permutation matrix.
+
+        A matrix is a permutation matrix if the sum of each row and column is exactly 1.
+        """
+        abs_matrix = np.absolute(self._matrix)
+        return (
+            abs_matrix.sum(
+                axis=0) == 1).all() and (
+            abs_matrix.sum(
+                axis=1) == 1).all()
+
+    def chip_to_stage(self, chip_coordinate):
```

Comment:
Is my assumption correct that this function is only correct for relative coordinates? If yes please update docstring accordingly.

You don't need to change anything, but we should keep in mind that for the full calibration it can come in handy to have `chip_to_stage` (and `stage_to_chip`) functions available for both, relative and absolute coordinates.